pkg/providers/v1/aws.go

```diff
@@ -1503,6 +1503,23 @@ func (c *Cloud) NodeAddresses(ctx context.Context, name types.NodeName) ([]v1.No
 			}
 		}
 
+		for _, macID := range macIDs {
+			ipv6Path := path.Join("network/interfaces/macs/", macID, "ipv6s")
+			internalIPv6s, err := c.metadata.GetMetadata(ipv6Path)
+			if err != nil {
```

Comment:
If we ignore all errors (like this) then an IMDS timeout/throttle will result in our IPv6 node addresses disappearing sometimes.  That seems undesirable.

I think this needs to be more explicit:
```golang
// isAWSNotFound returns true if the error was caused by an AWS API 404 response.
func isAWSNotFound(err error) bool {
	if err != nil {
		var aerr awserr.RequestFailure
		if errors.As(err, &aerr) {
			return aerr.StatusCode() == http.StatusNotFound
		}
	}
	return false
}

// ...
if isAWSNotFound(err) {
    internalIPv6s = nil  // 404 -> no IPv6 addresses
} else if err != nil {
    return nil, fmt.Errorf("error querying AWS metadata for %q: %q", ipv6Path, err)
}
```